build-tools/packages/build-cli/src/commands/generate/buildVersion.ts

```diff
@@ -102,13 +109,27 @@ export default class GenerateBuildVersionCommand extends BaseCommand<
 		}
 
 		// Generate and print the version to console
-		const simpleVersion = getSimpleVersion(
+		let simpleVersion = getSimpleVersion(
 			fileVersion,
 			flags.build,
 			isRelease,
 			useSimplePatchVersion,
 		);
-		const version = useTestVersion ? `0.0.0-${flags.build}-test` : simpleVersion;
+
+		if (isRelease && isAlphaOrBetaTypes) {
+			this.errorLog(`ERROR: alpha/beta prereleases are allowed`);
+		}
+
+		if (!isRelease && isAlphaOrBetaTypes) {
+			simpleVersion = `${simpleVersion}-${flags.packageTypes}-types`;
+		}
+
+		const version = useTestVersion
+			? flags.packageTypes === "alpha" || flags.packageTypes === "beta"
+				? `0.0.0-${flags.build}-${flags.packageTypes}-types-test`
+				: `0.0.0-${flags.build}-test`
+			: simpleVersion;
```

Comment:
Is this needed? Why treat test versions differently?

Reply:
When creating a test build with alpha or beta types selected, it's important to include the corresponding `alpha-types` or `beta-types` labels in the version. To ensure this, I've introduced specific test cases where the environment variable `TEST_BUILD` is set to `true`. If these labels are not added, the test build won't incorporate the alpha or beta keywords into its version number.